### ISSUE TITLE
How to access the contents of a Map from a template?

### ISSUE THREAD

**Issue by acrispino** [mbosecke/pebble#94]
Hello, I'd like to access the contents of a Map using a key obtained by looping over another collection. But I'm not sure how to accomplish that. I can't use the subscript notation ("Only strings and numbers allowed within square brackets") but I can't seem to use Map.get() either. 

I've included some contrived example code to show you what I'm after.

I'm using pebble 1.5.2

```Java
public class Application {
  public static void main(String[] args) throws IOException, PebbleException {
    PebbleEngine engine = new PebbleEngine();
    engine.setStrictVariables(true);
    PebbleTemplate compiledTemplate = engine.getTemplate("home.html");
    Map<String, Object> context = new HashMap<>();

    Map<Integer, User> usersById = new HashMap<>();
    usersById.put(1, new User("John", "Doe"));
    usersById.put(2, new User("Jane", "Doe"));
    context.put("usersById", usersById);

    List<Integer> userIds = new ArrayList<>();
    userIds.add(1);
    userIds.add(2);
    context.put("userIds", userIds);

    Writer writer = new StringWriter();
    compiledTemplate.evaluate(writer, context);
    System.out.println(writer.toString());
  }

  public static class User {
    private final String firstName;
    private final String lastName;

    public User(String firstName, String lastName) {
      this.firstName = firstName;
      this.lastName = lastName;
    }

    public String getFirstName() {
      return firstName;
    }

    public String getLastName() {
      return lastName;
    }
  }
}
```

```
<!DOCTYPE html>
<html>
<head>
  <title>TEST</title>
</head>
<body>
<h1>Users</h1>
<ul>
  {% for userId in userIds %}
    <li>{{ userId }}: {{ usersById.get(userId).getFirstName() }}</li>

    {# Not allowed #}
    {# <li>{{ userId }}: {{ usersById[userId].getFirstName() }}</li> #}
  {% endfor %}
</ul>
</body>
</html>
```

With the above code and template I get this exception:
```
Exception in thread "main" com.mitchellbosecke.pebble.error.AttributeNotFoundException: Attribute [get] of [java.util.HashMap] does not exist or can not be accessed and strict variables is set to true. (?:?)
	at com.mitchellbosecke.pebble.node.expression.GetAttributeExpression.evaluate(GetAttributeExpression.java:144)
    ...
```

**Comment by haducloc** [comment]
You should read document to know how to access Map. http://www.mitchellbosecke.com/pebble/documentation/guide/basic-usage

Try this: {{ usersById[userId].firstName }}

**Comment by acrispino** [comment]
I haven't been able to make it work that way:
```
Exception in thread "main" com.mitchellbosecke.pebble.error.ParserException: Only strings and numbers allowed within square brackets. (home.html:13)
```
I read that to mean that the subscript notation can only be used with literal strings and numbers, not with variables containing strings/numbers. Also, the example of subscript notation in the basic usage doc involves a literal string key.

**Comment by mbosecke** [comment]
I just released a new version of Pebble, v1.6.0, which I think might fix this issue. There was a bug in Pebble that prevented it from finding methods using reflection if the method had primitive argument types instead of boxed types. In this case, it's the "get" method it couldn't find.

**Comment by acrispino** [comment]
Both my example code and my project are working fine /w 1.6.0. Thanks for your help!
